wso2/carbon-mediation
# Fixes for cxf rm feature

This will change some class paths in cxf feature.

To test this please follow steps:

1. Add this feature through p2-repo
2. It will add cxf-bundle and jetty-bundle
3. Add relevant configuration file (make sure to change keystore paths in that file).
4. Use following configuration to add endpoint
```
<inboundEndpoint
        class="org.wso2.carbon.inbound.endpoint.ext.wsrm.InboundRMHttpListener"
        name="RM_INBOUND_NEW_EXT" onError="fault" sequence="RMIn" suspend="false">
        <parameters>
            <parameter name="inbound.cxf.rm.config-file">repository/conf/cxf/server.xml</parameter>
            <parameter name="inbound.behavior">listening</parameter>
            <parameter name="inbound.cxf.rm.host">127.0.0.1</parameter>
            <parameter name="coordination">true</parameter>
            <parameter name="sequential">true</parameter>
            <parameter name="inbound.cxf.rm.port">20940</parameter>
        </parameters>
    </inboundEndpoint>


<sequence name="RMIn" onError="fault">
        <in>
            <property name="PRESERVE_WS_ADDRESSING" value="true"/>
            <header action="remove" name="wsrm:Sequence" xmlns:wsrm="http://schemas.xmlsoap.org/ws/2005/02/rm"/>
            <header action="remove" name="wsa:To" xmlns:wsa="http://www.w3.org/2005/08/addressing"/>
            <header action="remove" name="wsa:FaultTo" xmlns:wsa="http://www.w3.org/2005/08/addressing"/>
            <log level="full"/>
            <send>
                <endpoint>
                    <address uri="http://localhost:9000/services/SimpleStockQuoteService"/>
                </endpoint>
            </send>
        </in>
        <out>
            <log level="custom">
                <property name="RM OutSequence" value="Sending Response"/>
            </log>
            <send/>
        </out>
    </sequence>

```
5. Build the SimpleStockQuote service in samples/axis2Server folder and start sample server.
6. Send the following request to test:
```
ant stockquote -Dsymbol=IBM -Dmode=quote -Daddurl=http://localhost:20940 -Dwsrm=true

```
